components/common-go/baseserver/options.go

```diff
@@ -99,6 +103,17 @@ func WithMetricsRegistry(r *prometheus.Registry) Option {
 	}
 }
 
+func WithHealthHandler(handler healthcheck.Handler) Option {
+	return func(cfg *config) error {
+		if handler == nil {
```

Comment:
The `handler` is not a pointer, do we have to add the nil check? Or, should we change to using a pointer?

Reply:
Handler is an interface, so the actual value is a pointer.